hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/TestContainerReportHandler.java

```diff
@@ -446,64 +484,70 @@ public void testClosingToClosedForECContainer()
   }
 
   /**
-   * Tests that a DELETING RATIS container transitions to CLOSED if a non-empty CLOSED replica is reported. It does not
-   * transition if a non-empty CLOSING replica is reported.
+   * Tests that a DELETING or DELETED RATIS container transitions to CLOSED if a non-empty replica in OPEN, CLOSING,
+   * CLOSED, QUASI_CLOSED or UNHEALTHY state is reported. It should not transition if the replica is in INVALID or
+   * DELETED states.
    */
-  @Test
-  public void ratisContainerShouldTransitionFromDeletingToClosedWhenNonEmptyClosedReplica() throws IOException {
-    ContainerInfo container = getContainer(LifeCycleState.DELETING);
+  @ParameterizedTest
+  @MethodSource("containerAndReplicaStates")
+  public void ratisContainerShouldTransitionFromDeletingOrDeletedToClosedWhenNonEmptyReplica(
```

Comment:
There is also `ecContainerShouldTransitionFromDeletingToClosedWhenNonEmptyClosedReplica` later in this class. We can probably just make EC/Ratis a parameter in this test as well. We would just have to skip quasi-closed replica state for the EC container when generating the combinations in `containerAndReplicaStates`.

Reply:
I checked/tried and it looks to me there are still subtle differences between these. I'll probably not going to bother refactoring this in this PR.

Reply:
The handling should be same regardless of replication type. Right now there is more coverage for the Ratis case but we need Ratis and EC to have the same coverage. Can you elaborate on what the difference is? We need to make sure it is not exposing a legitimate issue.

Reply:
ah i got your point. done.